pkg/controller/monitoring.go

```diff
@@ -120,6 +121,49 @@ func (c *Controller) updateHealthStatusForTenant(tenant *miniov2.Tenant) error {
 	if err != nil {
 		// show the error and continue
 		klog.Infof("'%s/%s' Failed to get cluster health: %v", tenant.Namespace, tenant.Name, err)
+		if strings.Contains(err.Error(), "failed to verify certificate") {
+			externalCertSecret := tenant.Spec.ExternalCertSecret
+			klog.Info("Let's check if there is an external cert for the tenant...")
+			if externalCertSecret != nil {
+				// Check that there is a secret that starts with "operator-ca-tls-" to proceed with the renewal
+				secretsAvailableAtOperatorNS, _ := c.kubeClientSet.CoreV1().Secrets(miniov2.GetNSFromFile()).List(context.Background(), metav1.ListOptions{})
+				for _, secret := range secretsAvailableAtOperatorNS.Items {
+					// Check if secret starts with "operator-ca-tls-"
+					secretName := OperatorCATLSSecretName + "-"
+					if strings.HasPrefix(secret.Name, secretName) {
+						klog.Infof("External secret found: %s", secret.Name)
+						klog.Info("We are going to renew the external certificate for the tenant...")
+						// Steps:
+						// 1. Delete the secret that starts with operator-ca-tls- because it is expired
+						c.kubeClientSet.CoreV1().Secrets(miniov2.GetNSFromFile()).Delete(context.Background(), secret.Name, metav1.DeleteOptions{})
+						// 2. Get the new certificate generated by cert-manager
+						tenantSecretName := tenant.Spec.ExternalCertSecret[0].Name
+						data, _ := c.kubeClientSet.CoreV1().Secrets(tenant.Namespace).Get(context.Background(), tenantSecretName, metav1.GetOptions{})
+						CACertificate := data.Data["0"]
```

Comment:
Should we also check if data.Data length is > 0? so that there's no exceptions there either.